logcat/src/commonMain/kotlin/logcat/Logcat.kt

```diff
@@ -91,3 +91,71 @@ inline fun logcat(
     }
   }
 }
+
+/**
+ * An alternative way of calling the free [logcat] method using a Timber-like API.
+ *
+ * Use like:
+ *
+ * ```kotlin
+ * import logcat.logcat
+ *
+ * class MouseController {
+ *
+ *   fun play {
+ *     var state = "CHEEZBURGER"
+ *     logcat.d { "I CAN HAZ $state?" }
+ *     // logcat output: D/MouseController: I CAN HAZ CHEEZBURGER?
+ *
+ *     logcat.i { "DID U ASK 4 MOAR INFO?" }
+ *     // logcat output: I/MouseController: DID U ASK 4 MOAR INFO?
+ *
+ *     logcat.w { exception.asLog() }
+ *     // logcat output: W/MouseController: java.lang.RuntimeException: FYLEZ KERUPTED
+ *     //                        at sample.MouseController.play(MouseController.kt:22)
+ *     //                        ...
+ *
+ *     logcat.e("Lolcat") { "OH HI" }
+ *     // logcat output: E/Lolcat: OH HI
+ *   }
+ * }
+ * ```
+ */
+@Suppress("ClassName")
+object logcat {
+  context(subject: Any)
+  fun v(
+    tag: String? = null,
+    message: () -> String
+  ) = subject.logcat(LogPriority.VERBOSE, tag, message)
+
+  context(subject: Any)
+  fun d(
```

Comment:
Part of the design of logcat is that inlining means we completely skip any method invocation is logging if off. We're losing that here.